packages/components/src/components/form-progress/form-progress.jsx

```diff
@@ -3,82 +3,73 @@ import PropTypes from 'prop-types';
-                                    <Text
-                                        as='p'
-                                        align='center'
-                                        size='xxs'
-                                        weight={active ? 'bold' : 'unset'}
-                                        color={active ? 'loss-danger' : 'prominent'}
-                                    >
-                                        {item.header.title}
-                                    </Text>
-                                </div>
-                            </React.Fragment>
-                        ))}
-                        <div ref={this.el_completed_bar} className='dc-form-progress__steps--after' />
-                    </div>
+    return (
+        <div className='dc-form-progress'>
+            <div className='dc-form-progress__header'>
+                <Text as='h2' color='prominent' weight='bold' line_height='unset'>
+                    {steps[current_step].header.active_title}
+                </Text>
+                <div className='dc-form-progress__steps'>
+                    <div
+                        className='dc-form-progress__steps--before'
+                        style={{
+                            width: `calc(100% * ${steps.length - 1} / ${steps.length})`,
+                        }}
+                    />
+                    {steps.map((item, idx) => (
+                        <React.Fragment key={idx + 1}>
+                            {(active = idx === current_step)}
+                            <div className='dc-form-progress__step'>
+                                <Text
+                                    align='center'
+                                    size='xs'
+                                    weight='bold'
+                                    color='colored-background'
+                                    className={classNames('identifier', {
+                                        'identifier--active': idx <= current_step,
+                                    })}
+                                >
+                                    {idx + 1}
+                                </Text>
+                                <Text
+                                    as='p'
+                                    align='center'
+                                    size='xxs'
+                                    weight={active ? 'bold' : 'unset'}
+                                    color={active ? 'loss-danger' : 'prominent'}
+                                >
+                                    {item.header.title}
+                                </Text>
+                            </div>
+                        </React.Fragment>
+                    ))}
+                    <div ref={el_completed_bar} className='dc-form-progress__steps--after' />
                 </div>
             </div>
-        );
-    }
-}
+        </div>
+    );
+};
 
 FormProgress.propTypes = {
```

Comment:
Are we using all these props?

Reply:
It looks like we don't use, was just worried why they are still there, then will remove them, thx!